### PR TITLE
Fix bugs in BCF I/O

### PR DESCRIPTION
**Pull request by alumi**
#### Summary
This PR fixes miscellaneous bugs in BCF reader/writer

1. Header string is missing a `\newline` at its end
2. Duplicated `idx`s of FILTER/INFO/FORMAT are not allowed
3. Some integer values are reserved: `0x80–0x87`, `0x8000–0x8007`, `0x80000000–0x80000007`
4. Incorrect values are written for MISSING float `0x7F800001`
5. Encoding/decoding is broken for some types and values
6. `END` info is not referred to when encoding reference length

#### Tests

- `lein check` 🆗
- `lein test :all` 🆗
- `lein cloverage` 🆗
- `lein eastwood` 🆗

### CODE REVIEW

File: src/cljam/io/bcf/writer.clj [hunk 1]
```diff
@@ -98,104 +110,117 @@
...
-  [kvs]
-  (let [bas (->> kvs
-                 (mapcat
-                  (fn [[k v]]
-                    [(encode-typed-value k)
-                     (encode-typed-value v)])))
-        bb (ByteBuffer/allocate (reduce + (map alength bas)))]
-    (doseq [ba bas]
-      (.put bb ^bytes ba))
+(defn- merge-bytes
```
Comment (athos) [inline src/cljam/io/bcf/writer.clj]: It's just my personal opinion, but how about naming this function `concat-bytes`? The word `merge` implies (at least to me) the existence of a certain underlying merging strategy.

**Comment by alumi** [comment]
@athos Thank you for the comments! I've fixed them in c8172d7...5f030a7

**Comment by athos** [comment]
LGTM 👍I'll merge it.

**Comment by alumi** [comment]
Thank you for reviewing! @athos @yito88 😃

**Pull request merged**